### PR TITLE
Added logs, exit on error

### PR DESCRIPTION
**Pull request by Mingela**

### CODE REVIEW

**Review by Alexey-N-Chernyshov — APPROVED** [review]

File: brvs-core/src/main/java/iroha/validation/listener/BrvsIrohaChainListener.java [hunk 1]
```diff
@@ -75,14 +79,27 @@ public BrvsIrohaChainListener(
...
 
+  /**
+   * Returns a relevant query api instance
+   *
+   * @param accountId user that transactions should be queried for
+   * @param keyPair user keypair
+   * @return user {@link QueryAPI} instance
+   */
+  private QueryAPI getQueryApiFor(String accountId, KeyPair keyPair) {
+    if (!queryAPIMap.containsKey(accountId)) {
```
Comment (dolgopolovwork) [inline brvs-core/src/main/java/iroha/validation/listener/BrvsIrohaChainListener.java:97]: A race condition may happen here.

File: brvs-core/src/main/java/iroha/validation/transactions/provider/impl/BasicTransactionProvider.java [hunk 2]
```diff
@@ -102,17 +104,22 @@ public BasicTransactionProvider(
...
-              // if only BRVS signatory remains
-              if (isBatchSignedByUsers(transactionBatch)) {
-                if (savedMissingInStorage(transactionBatch)) {
-                  cacheProvider.put(transactionBatch);
+    try {
+      irohaReliableChainListener
+          .getAllPendingTransactions(registrationProvider.getRegisteredAccounts())
+          .forEach(transactionBatch -> {
+                // if only BRVS signatory remains
+                if (isBatchSignedByUsers(transactionBatch)) {
```
Comment (dolgopolovwork) [inline brvs-core/src/main/java/iroha/validation/transactions/provider/impl/BasicTransactionProvider.java]: I would rewrite it like that `if (isBatchSignedByUsers(transactionBatch) && savedMissingInStorage(transactionBatch))`

File: brvs-core/src/main/java/iroha/validation/listener/BrvsIrohaChainListener.java (hunk 1 above)
Comment (Mingela) [reply brvs-core/src/main/java/iroha/validation/listener/BrvsIrohaChainListener.java:97]: Oh ok

File: brvs-core/src/main/java/iroha/validation/transactions/provider/impl/BasicTransactionProvider.java (hunk 2 above)
Comment (Mingela) [reply brvs-core/src/main/java/iroha/validation/transactions/provider/impl/BasicTransactionProvider.java]: ok

**Review by dolgopolovwork — APPROVED** [review]

**Pull request merged**